docs/oid-info.md

```diff
@@ -58,11 +60,89 @@ the git ref that the workflow run was based upon.
+| iss                | iss     | iss       | iss        | Issuer                       | This already exists. For example: https://token.actions.githubusercontent.com                                                                                                          |
+| exp                | exp     | exp       | exp        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
+| nbf                | nbf     | nbf       | nbf        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
+| iat                | iat     | iat       | iat        | N/A                          | Only used to validate the JWT.                                                                                                                                                         |
+| repository         | project_path     | ??       | ??        | Source URI                   | Should include a fully qualified repository URL.                                                                                                                                       |
+| sha                | ??     | ??       | build_commit | Source Digest                | An immutable reference to a specific version of the source code. Should include the digest type followed by the digest, e.g. `sha1:abc123`.                                            |
+| ref                | ref     | ??       | build_branch | Source Ref                   | The source ref that the build run was based upon. For example: refs/head/main.                                                                                                         |
+| repository_id         | project_id     | ??       | ??        | Source Identifier     | Stable identifier for the owner of the source repository.                                                                                                                                      |
+| repository_owner         | namespace_path     | ??       | ??        | Source Owner URI     | Fully qualified URL for the owner of the source repository.                                                                                                                                     |
+| repository_owner_id         | namespace_id     | ??       | ??        | Source Owner Identifier     | Stable identifier for the owner of the source repository.                                                                                                                                   |
```

Comment:
Q: For providers like Buildkite and Circle who don't also host the source, should they forward the underlying source/source owner ids in their workload identity? How do we track the "build identity" that executed the build and distinguish this from the source identity? Is the build config URI enough?

Reply:
Someone more familiar with those CI services can probably answer better than me, but my expectation is that `repository_id` would be the be the "project" or "pipeline" ID on Buildkite or Circle, while `repository_owner_id` would be the ID of the `User` or other owner model that configured the pipeline.

That's contingent on these services having some kind of underlying relational design that looks roughly like the following:

```
User(id, ...) from table of USERS
Project(id, user_id_roles) from table of PROJECTS
```

...but I think that's probably a reasonable assumption 🙂

Reply:
Yeah would be good to get some clarity here. I was imaging `Source *` extensions would all point to the source repository. So if you use Circle to build with code on GitHub, Source URI would point to the github repo. Ref, digest would also point to the repo.

The reason we would want Source Owner Identifier and Source Identifier would be to spot transfers/resurrected repositories, in this example this would be the GitHub repo. Given this, I think we actually want Circle et al to provide these source/source owner ids in their OIDC claims. 

Build Config URI would presumably, in most cases, point to a build config file in the same repo. 

The Builder Signer URI, could in Circle's case be a [Circle Orb](https://circleci.com/orbs/) if one is used, otherwise this would point to the same above build config uri if one was not used.

Given this, there's actually no space in the current extensions for any build system identifiers like "project" or "pipeline" ID on Buildkite or Circle 🤔 

Question: do we need something specific here to track separate build system/build owner identifiers when this differs from the source system?

cc @haydentherapper what are your thoughts on above?

Reply:
We should have a way to differentiate CI platforms, that should be a part of what’s validated by the verifier. Otherwise I could imagine a threat vector where an attacker creates a project on a different platform matching all of the project/repo details, but changing the source.

Reply:
> We should have a way to differentiate CI platforms, that should be a part of what’s validated by the verifier.

This makes sense, is this not already covered by the OIDC issuer extension?

Reply:
Yes this is! OIDC issuer is sufficient

Reply:
> but my expectation is that `repository_id` would be the be the "project" or "pipeline" ID on Buildkite or Circle, while `repository_owner_id` would be the ID of the `User` or other owner model that configured the pipeline.

I am familiar with Buildkite. Reading this spec, I would assume any OID mentioning "Source" or "Repository" would be some pointer back to, or meta data about the GitHub (in my case) repository and commit/ref that triggered the build.

And I'd expect `Run Invocation URI` to be something like `https://buildkite.com/mycompany/mypipeline/builds/118`

And my issuer (Buildkite) would be trusted to set the appropriate fields related to the source repository.